openshift/message-board/message-service/src/test/java/org/hibernate/demo/message/post/it/MessageTaskIT.java

```diff
@@ -0,0 +1,41 @@
+package org.hibernate.demo.message.post.it;
+
+import java.io.File;
+
+import org.hibernate.demo.message.post.util.DeploymentUtil;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.TargetsContainer;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
+
+@RunWith(Arquillian.class)
+public class MessageTaskIT {
+
+	@Deployment(name = "wildfly")
+	@TargetsContainer("wildfly")
+	public static WebArchive doDeploy() {
+		return DeploymentUtil.create();
+	}
+
+	@Deployment(name = "infinispan", testable = false)
+	@TargetsContainer("infinispan")
+	public static JavaArchive getInfinispanDeployment() {
+		File file = Maven.resolver()
+			.resolve( "org.hibernate.demos.messageboard:message-server-task:jar:1.0.0-SNAPSHOT" )
```

Comment:
Can we read the version from somewhere? I think in OGM we have something for this

Reply:
In Search we just use a resource file, configure Maven to filter it, and read the file when executing tests.

See:

* https://github.com/hibernate/hibernate-search/blob/master/legacy/integrationtest/wildfly/src/test/resources/module-versions.properties
* https://github.com/hibernate/hibernate-search/blob/master/legacy/integrationtest/wildfly/src/test/java/org/hibernate/search/test/integration/VersionTestHelper.java#L113
* https://github.com/hibernate/hibernate-search/blob/master/legacy/integrationtest/wildfly/pom.xml#L229

Reply:
You're right @DavideD and thank you @yrodiere for the suggestion, it makes sense for me. I applied the same strategy here https://github.com/infinispan/infinispan/pull/6175.